File: iPhone/Apex Cougars/Classes/TeamDetailService.m
//
//  TeamDetailService.m
//  iAthletics Framework
//
//  Created by Robert Signore on 6/17/10.
//  Copyright 2010 Appapro. All rights reserved.
//

#import "TeamDetailService.h"


@implementation TeamDetailService

@synthesize teamInfo;
@synthesize currentLink;
@synthesize newsLinks;

#pragma mark Constructors/Destructors

-(id) initWithApplication:(iAthleticsApplication *)theApp
{
	self = [super initWithApplication:theApp];
	
	TeamInfo *newInfo = [[TeamInfo alloc]init];
	
	self.teamInfo = newInfo;
	[newInfo release];
	
	NSMutableArray *newArray = [[NSMutableArray alloc]initWithCapacity:20];
	self.newsLinks = newArray;
	[newArray release];
	
	return self;
}

-(void)dealloc
{
	[self.teamInfo release];
	[self.currentLink release];
	[self.newsLinks release];
	
	[super dealloc];
}

#pragma mark service request methods

-(TeamInfo *)requestServiceWithTeamId:(NSInteger)teamId
{
	// create the parameter string
	//
	NSString *paramString = [NSString stringWithFormat:@"teamId=%d", teamId];
	[self requestServiceName:@"GetTeamInfo" parameters:paramString];
	
	if([newsLinks count] > 0)
	{
		self.teamInfo.newsLinks = [NSArray arrayWithArray:self.newsLinks];
	}
	
	return self.teamInfo;		
}

#pragma mark -
#pragma mark iAthleticsService ABSTRACT methods

-(void) elementStart:(NSString *)elementName
{
	// prepare to receive new AppSeason Info
	//
	if([elementName compare:@"NewsLink"] == NSOrderedSame)
	{
		NewsLink *newLink = [[NewsLink alloc]init];
		self.currentLink = newLink;
		[newLink release];
	}
}

-(void)elementEnd:(NSString *)elementName withText:(NSString *)text
{
	
	// team id
	//
	if([elementName compare:@"TeamId"] == NSOrderedSame)
	{
		self.teamInfo.teamId = [text intValue];
	}
	
	// overall results
	//
	else if([elementName compare:@"TeamOverallResults"] == NSOrderedSame)
	{
		self.teamInfo.overallResults = text;
	}	
	// sport name
	//
	else if([elementName compare:@"SportName"] == NSOrderedSame)
	{
		self.teamInfo.sportName = text;
	}	
	
	// team name
	//
	else if([elementName compare:@"TeamName"] == NSOrderedSame)
	{
		self.teamInfo.teamName = text;
	}
	
	// team photo?
	//
	else if([elementName compare:@"TeamPhotoUrl"] == NSOrderedSame)
	{
		iAthleticsImage *newImage = [[iAthleticsImage alloc]initWithUrl:text];
		self.teamInfo.photo = newImage;
		[newImage release];
	}

	/////
	//
	// parsing for the NewsLinks in the message
	//
	/////
	
	// link name
	//
	else if([elementName compare:@"LinkName"] == NSOrderedSame)
	{
		self.currentLink.linkName = text;
	}
	
	// link description
	//
	else if([elementName compare:@"LinkDescription"] == NSOrderedSame)
	{
		self.currentLink.linkDescription = text;
	}
	
	// the link URL
	//
	else if([elementName compare:@"LinkUrl"] == NSOrderedSame)
	{
		self.currentLink.linkUrl = text;
	}
	
	// the icon URL?
	//
	else if([elementName compare:@"LinkIconUrl"] == NSOrderedSame)
	{
		iAthleticsImage *newImage = [[iAthleticsImage alloc]initWithUrl:text];		
		self.currentLink.image = newImage;
		[newImage release];
	}
	
	else if([elementName compare:@"LinkTypeId"] == NSOrderedSame)
	{
		self.currentLink.linkType = [text intValue];
	}
	
	// if this is the end of the NewsLink then add the currentLink to the 
	// links array
	else if([elementName compare:@"NewsLink"] == NSOrderedSame)
	{
		[self.newsLinks addObject:self.currentLink];
	}
}


@end

File: Android/iAthletics/src/com/appapro/iathletics/services/TeamDetailService.java
package com.appapro.iathletics.services;

import com.appapro.iathletics.activities.iAthleticsActivity;
import com.appapro.iathletics.types.NewsLink;
import com.appapro.iathletics.types.TeamInfo;
import com.appapro.iathletics.types.iAthleticsImage;

public class TeamDetailService extends iAthleticsService
{
	TeamInfo teamInfo = new TeamInfo();
	NewsLink currentLink;

	public TeamDetailService(iAthleticsActivity activity)
	{
		super(activity);
	}

	public TeamInfo getTeamInfo(int teamId)
	{
		// create the parameter string
		//
		String parameters = "teamId=" + Integer.toString(teamId);

		this.requestService("GetTeamInfo", parameters);
		
		return this.teamInfo;		
	}

	@Override
	protected void elementStart(String elementName)
	{
		// prepare to receive new news link Info
		//
		if(elementName.compareTo("NewsLink") == 0)
		{
			this.currentLink = new NewsLink();
		}
	}

	@Override
	protected void elementEnd(String elementName, String text)
	{
		
		// team id
		//
		if(elementName.compareTo("TeamId") == 0)
		{
			this.teamInfo.setTeamId(Integer.valueOf(text));
		}
		
		// overall results
		//
		else if(elementName.compareTo("TeamOverallResults") == 0)
		{
			this.teamInfo.setOverallResults(text);
		}	
		// sport name
		//
		else if(elementName.compareTo("SportName") == 0)
		{
			this.teamInfo.setSportName(text);
		}	
		
		// team name
		//
		else if(elementName.compareTo("TeamName") == 0)
		{
			this.teamInfo.setTeamName(text);
		}
		
		// team photo?
		//
		else if(elementName.compareTo("TeamPhotoUrl") == 0)
		{
			this.teamInfo.setPhoto(new iAthleticsImage(text, this.getActivity()));
		}

		/////
		//
		// parsing for the NewsLinks in the message
		//
		/////
		
		// link name
		//
		else if(elementName.compareTo("LinkName") == 0)
		{
			this.currentLink.setLinkName(text);
		}
		
		// link description
		//
		else if(elementName.compareTo("LinkDescription") == 0)
		{
			this.currentLink.setLinkDescription(text);
		}
		
		// the link URL
		//
		else if(elementName.compareTo("LinkUrl") == 0)
		{
			this.currentLink.setLinkUrl(text);
		}
		
		// the icon URL?
		//
		else if(elementName.compareTo("LinkIconUrl") == 0)
		{
		
			this.currentLink.setImage(new iAthleticsImage(text, this.getActivity()));
		}
		
		else if(elementName.compareTo("LinkTypeId") == 0)
		{
			this.currentLink.setLinkType(Integer.valueOf(text));
		}
		
		// if this is the end of the NewsLink then add the currentLink to the 
		// links array
		else if(elementName.compareTo("NewsLink") == 0)
		{
			this.teamInfo.addNewsLink(this.currentLink);
		}
	}

}
